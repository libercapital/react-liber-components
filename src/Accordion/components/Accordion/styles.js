import styled from 'styled-components';
import { media } from '../../../Util';
import { handleThemeFromObject } from '../../../Themes';

export const AccordionBox = styled.div`
  overflow: hidden;
  width: 100%;
  background-color: ${props =>
    props.open
      ? handleThemeFromObject(props.theme, 'accordion.boxBackgroundOpen', '#fff')
      : handleThemeFromObject(props.theme, 'accordion.boxBackgroundClosed', '#f9f9f9')};
  border: ${props =>
    props.open
      ? `1px solid ${handleThemeFromObject(props.theme, 'accordion.boxBorderOpen', '#fff')}`
      : `solid 1px ${handleThemeFromObject(
          props.theme,
          'accordion.boxBorderClosed',
          'rgba(150, 150, 150, 0.18)',
        )}`};
  transition: all 400ms ${props => (props.open ? 'ease-in' : 'ease-out')};
  &:hover {
    background-color: ${props =>
      props.open
        ? handleThemeFromObject(props.theme, 'accordion.boxBackgroundOpen', '#fff')
        : handleThemeFromObject(props.theme, 'accordion.boxBackgroundHover', '#f3f3f3')};
  }
`;

export const ShowBox = styled.div`
  max-height: ${props => (props.open ? props.size : '0px')};
  transition: max-height 400ms ${props => (props.open ? 'ease-in' : 'ease-out')};
  overflow: hidden;
`;

export const AccordionAction = styled.div`
  padding: 20px 50px;
  cursor: pointer;
  box-sizing: border-box;
  display: inline-block;
  background-color: ${props => (props.open ? props.color || 'transparent' : 'transparent')};
  width: 100%;
  transition: all 400ms ${props => (props.open ? 'ease-in' : 'ease-out')};
  position: relative;

  ${media.large`
    padding: 30px 50px;
  `}

  &:after {
    position: absolute;
    top: calc(50% - 4px);
    left: 15px;
    width: 0;
    height: 0;
    padding: 0;
    content: '';
    border-left: 7px solid transparent;
    border-right: 7px solid transparent;
    border-top: 8px solid
      ${props =>
        props.open
          ? handleThemeFromObject(props.theme, 'accordion.actionBorderTopOpen', '#009dff')
          : handleThemeFromObject(props.theme, 'accordion.actionBorderTopClosed', '#4e4e4e')};
    pointer-events: none;
    transform: ${props => (props.open ? 'rotate(0deg)' : 'rotate(-90deg)')};
    transition: all 400ms ${props => (props.open ? 'ease-in' : 'ease-out')};
  }
`;
