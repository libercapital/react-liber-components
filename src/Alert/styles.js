import styled, { css } from 'styled-components';
import { MdClose } from '../Icons/MaterialDesign';

const icon = css`
  --size: 24px;
  min-width: var(--size);
  max-width: var(--size);
  min-height: var(--size);
  max-height: var(--size);
`;

export const CloseIcon = styled(MdClose)`
  ${icon}
  &:hover {
    cursor: pointer;
  }
`;

export const AlertBox = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  width: 100%;
`;

export const AlertContent = styled.div`
  display: flex;
  flex-grow: 1;
  flex-direction: column;
  padding: 24px 0;
`;

export const AlertInfoContent = styled.div`
  font-family: Roboto;
  font-size: 16px;
  font-weight: 500;
  line-height: 1.38;
  letter-spacing: normal;
  color: #405f71;
`;
export const AlertAdditionalInfoContent = styled.div`
  font-family: Roboto;
  font-size: 16px;
  font-weight: normal;
  line-height: 1.38;
  letter-spacing: 0px;
  color: #587383;
  padding-top: 4px;
`;

export const AlertIcon = styled.div`
  display: flex;
  flex-grow: 0;
  align-items: center;
  padding: 24px;
`;
