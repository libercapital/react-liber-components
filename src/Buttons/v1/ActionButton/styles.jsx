import styled from 'styled-components';
import { media } from '../../../Util';


export const Label = styled.div`
  text-align: center;
  width: unset;
  opacity: 1;
  transition: all 300ms  ${props => (props.hover ? 'ease-out' : 'ease-in')};
  white-space: nowrap;
  color: inherit;

  ${media.large`
    width: ${props => (props.hover ? props.size || '0px' : '0px')};
    opacity: ${props => (props.hover ? '1' : '0')};
  `}
  ${media.extraLarge`
    width: ${props => (props.hover ? props.size || '0px' : '0px')};
    opacity: ${props => (props.hover ? '1' : '0')};
  `}
`;

export const Icon = styled.div`
  margin-right: 15px;
  transition: margin 300ms  ${props => (props.hover ? 'ease-out' : 'ease-in')};
  ${media.large`
    margin-right: ${props => (props.hover ? '15px' : '0px')};
  `}
  ${media.extraLarge`
    margin-right: ${props => (props.hover ? '15px' : '0px')};
  `}
`;
