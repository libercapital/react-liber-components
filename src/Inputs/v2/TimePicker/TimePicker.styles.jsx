import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';
import ScrollArea from 'react-perfect-scrollbar';
import { TextField } from '../InputField';
import { handleThemeFromObject as getTheme } from '../../../Themes';
import { MdClockOutline, MdCloseCircle } from '../../../Icons';


export const Container = styled.div`
  position: relative;
`;

export const Field = styled(TextField)`
  padding-right: 52px;
`;

export const Icon = styled(MdClockOutline)`
  --size: 20px;
  min-width: var(--size);
  max-width: var(--size);
  min-height: var(--size);
  max-height: var(--size);
  fill: #8c9fa9;
  position: absolute;
  right: 16px;
  top: calc(50% - 5px);
  pointer-events: none;
  opacity: ${({ hide }) => (hide ? '0' : '1')};
  transition: opacity 250ms cubic-bezier(0.23,1,0.32,1) 0ms;
`;

export const TimeContainer = styled.div`
  display: flex;
  justify-content: flex-start;
  align-items: stretch;
  height: var(--clock-container-size, 264px);
  & > div{
    flex-grow: 1;
    padding: 12px 0px;
    height: 100%;
    border-right: 1px solid ${props => getTheme(props.theme, 'colors.border', '#e8ebee')};
    box-sizing: border-box;
  }
  & > div:last-child{
    border-right: 0px;
  }
`;


export const Col = styled(ScrollArea)`
  display: flex;
  flex-direction: column;
  margin: 12x 0px;
  height: 100%;
  overflow: auto;
  flex-grow: 1;
  max-height: 100%;


  & .ps__rail-y{
    margin-top: 4px;
    margin-bottom: 4px;
  }

  & .ps__thumb-y{
    background-color: #e2e2e2!important;
    right: 4px;
  }

  & .ps__rail-y:hover, & .ps--clicking{
    background-color: transparent!important;
  }

  & .ps__rail-y:hover .ps__thumb-y,  & .ps__rail-y.ps--clicking .ps__thumb-y{
    width: 8px;
  }

   & .ps__rail-y{
    z-index: 10;
  }
`;

Col.propTypes = {
  children: PropTypes.oneOfType([PropTypes.any]),
};

Col.defaultProps = {
  children: '',
  options: {
    swipeEasing: true,
  },
};

const select = (selected) => {
  if (selected) {
    return css`
      color: ${props => getTheme(props.theme, 'colors.dark-text', '#405f71')};
      background-color: ${props => getTheme(props.theme, 'colors.header', '#f9fafa')};
  `;
  }
  return css`
    color: ${props => getTheme(props.theme, 'colors.dark-text70', '#798e9b')};
  `;
};

const body = css`
  font-size: 16px;
  font-weight: 400;
  line-height: 22px;
`;

export const Unit = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  min-height: 40px;
  width: 100%;
  transition:
    background-color 250ms cubic-bezier(0.23,1,0.32,1) 0ms,
    color 250ms cubic-bezier(0.23,1,0.32,1) 0ms;
  cursor: pointer;
  ${props => getTheme(props.theme, 'typography.body', body)};
  ${props => select(props.selected)}

  &:hover{
    background-color: ${props => getTheme(props.theme, 'colors.secondary', '#f0f2f4')};
  }
  &:nth-last-child(3){
    margin-bottom: var(--last-unit-margin, 200px);
  }
`;


const closeShow = css`
  opacity: 1;
  pointer-events: all;
`;
export const Close = styled(MdCloseCircle)`
  --size: 20px;
  min-width: var(--size);
  max-width: var(--size);
  min-height: var(--size);
  max-height: var(--size);
  fill: #8c9fa9;
  position: absolute;
  right: 16px;
  top: calc(50% - 5px);
  opacity: 0;
  pointer-events: none;
  ${({ show }) => show && closeShow}
  transition: opacity 250ms cubic-bezier(0.23,1,0.32,1) 0ms;
  z-index: 2;
  cursor: pointer;
`;
