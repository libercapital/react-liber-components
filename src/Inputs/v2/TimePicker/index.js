export { default } from './TimePicker';
export { default as Clock } from './Clock';
export { TimeContainer } from './TimePicker.styles';
