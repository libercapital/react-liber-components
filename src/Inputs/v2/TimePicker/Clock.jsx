import React, { PureComponent as Component } from 'react';
import PropTypes from 'prop-types';
import { pad } from '../../../Util';
import { TimeContainer, Col, Unit } from './TimePicker.styles';


const hours = [...Array(24)].map(Number.call, Number);

const minutes = [...Array(60)].map(Number.call, Number);

const seconds = minutes;


class Clock extends Component {
  units = {
    hour: {},
    min: {},
    sec: {},
  }

  col = {}

  componentDidUpdate() {
    const { hour: propHour, min: propMin, sec: propSec } = this.props;
    const hour = parseInt(propHour, 10);
    const min = parseInt(propMin, 10);
    const sec = parseInt(propSec, 10);
    if (this.units.hour[hour]) {
      this.col.hour.scrollTop = this.units.hour[hour].offsetTop;
    }
    if (this.units.min[min]) {
      this.col.min.scrollTop = this.units.min[min].offsetTop;
    }
    if (this.units.sec[sec]) {
      this.col.sec.scrollTop = this.units.sec[sec].offsetTop;
    }
  }

  clickCallback = (obj) => {
    const { onClick } = this.props;
    const { hour, min, sec } = obj;
    onClick(hour, min, sec, `${pad(hour)}:${pad(min)}:${pad(sec)}`);
  }

  handleClick = (key, value) => () => {
    const { hour, min, sec } = this.props;
    const obj = {
      hour: hour ? parseInt(hour, 10) : 0,
      min: min ? parseInt(min, 10) : 0,
      sec: sec ? parseInt(sec, 10) : 0,
      ...{ [key]: value },
    };
    this.clickCallback(obj);
  }

  addRef = (key, value) => (element) => {
    this.units[key][value] = element;
  }

  addColRef = col => (element) => {
    this.col[col] = element;
  }

  mapHours = selected => hour => (
    <Unit
      ref={this.addRef('hour', hour)}
      key={`hour_${hour}`}
      onClick={this.handleClick('hour', hour)}
      selected={hour === parseInt(selected, 10)}
    >{pad(hour)}
    </Unit>
  );

  mapMinutes = selected => min => (
    <Unit
      ref={this.addRef('min', min)}
      key={`min_${min}`}
      onClick={this.handleClick('min', min)}
      selected={min === parseInt(selected, 10)}
    >{pad(min)}
    </Unit>
  );

  mapSeconds = selected => sec => (
    <Unit
      key={`min_${sec}`}
      ref={this.addRef('sec', sec)}
      onClick={this.handleClick('sec', sec)}
      selected={sec === parseInt(selected, 10)}
    >{pad(sec)}
    </Unit>
  );


  render() {
    const {
      hour, min, sec,
    } = this.props;
    return (
      <TimeContainer>
        <div>
          <Col containerRef={this.addColRef('hour')}>{hours.map(this.mapHours(hour))}</Col>
        </div>
        <div>
          <Col containerRef={this.addColRef('min')}>{minutes.map(this.mapMinutes(min))}</Col>
        </div>
        <div>
          <Col containerRef={this.addColRef('sec')}>{seconds.map(this.mapSeconds(sec))}</Col>
        </div>
      </TimeContainer>
    );
  }
}

Clock.propTypes = {
  hour: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
  min: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
  sec: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
  onClick: PropTypes.func,
};

Clock.defaultProps = {
  hour: null,
  min: null,
  sec: null,
  onClick: () => {},
};

export default Clock;
