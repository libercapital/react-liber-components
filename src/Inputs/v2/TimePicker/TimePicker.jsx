import React, { PureComponent as Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import {
  Container, Field, Icon, Close,
} from './TimePicker.styles';
import Clock from './Clock';

import { SimpleDropDown } from '../../../DropDown';
import { ClickOutside } from '../../../Util';
import { PsGlobalStyle } from '../../../Assets';

const breakTime = (time) => {
  let [
    hour = '',
    min = '',
    sec = '',
  ] = time.split(':');
  hour = parseInt(hour, 10) > 23 ? '23' : hour;
  min = parseInt(min, 10) > 59 ? '59' : min;
  sec = parseInt(sec, 10) > 59 ? '59' : sec;

  return {
    time: time ? `${hour}:${min}:${sec}` : '', hour, min, sec,
  };
};

class TimePicker extends Component {
  state = {
    open: false,
    clear: false,
    hour: '',
    min: '',
    sec: '',
    time: '',
  }

  static getDerivedStateFromProps(props, state) {
    if (props.time) {
      return {
        ...breakTime(props.time),
      };
    }
    return state;
  }

  handleFocus = () => {
    this.setState({ open: true });
  }

  handleOutsideClick = () => {
    this.setState({ open: false });
  }


  onChangeCallback = () => {
    const { onChange } = this.props;
    const {
      time, hour, min, sec,
    } = this.state;
    onChange(time, hour, min, sec);
  }

  handleClick = (hour, min, sec, time) => {
    this.setState({
      time, hour, min, sec,
    }, this.onChangeCallback);
  }

  handleChange = (time) => {
    this.setState({
      ...breakTime(time),
    }, this.onChangeCallback);
  }

  handleOnMouseEnter = () => {
    const { time } = this.state;
    if (time.trim() !== '') { this.setState({ clear: true }); }
  }

  handleOnMouseLeave = () => {
    const { time } = this.state;
    if (time.trim() !== '') this.setState({ clear: false });
  }

  clearInput = () => {
    this.setState({
      hour: '',
      min: '',
      sec: '',
      time: '',
      clear: false,
    });
  }

  render() {
    const {
      time, open, hour, min, sec, clear,
    } = this.state;
    const {
      onChange, ...props
    } = this.props;
    return (
      <Fragment>
        <PsGlobalStyle />
        <ClickOutside onOutsideClick={this.handleOutsideClick}>
          <Container>
            <SimpleDropDown
              open={open}
              customButton={(
                <Field
                  width="340px"
                  {...props}
                  value={time}
                  version="2"
                  onChange={this.handleChange}
                  mask="99:99:99"
                  maskChar={null}
                  onFocus={this.handleFocus}
                  onMouseEnter={this.handleOnMouseEnter}
                  onMouseLeave={this.handleOnMouseLeave}
                />
          )}
            >
              <Clock onClick={this.handleClick} hour={hour} min={min} sec={sec} />
            </SimpleDropDown>
            <Icon hide={clear} />
          <Close // eslint-disable-line
            show={clear}
            onMouseEnter={this.handleOnMouseEnter}
            onClick={this.clearInput}
          />
          </Container>
        </ClickOutside>
      </Fragment>
    );
  }
}


TimePicker.propTypes = {
  label: PropTypes.string,
  time: PropTypes.string,
  onChange: PropTypes.func,
};

TimePicker.defaultProps = {
  label: '',
  time: null,

  onChange: () => {},
};

export default TimePicker;
