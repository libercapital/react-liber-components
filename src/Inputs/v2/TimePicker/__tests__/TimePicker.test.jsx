import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import sinon from 'sinon';
import { SimpleDropDown } from '../../../../DropDown';
import { ClickOutside } from '../../../../Util';
import { Field } from '../TimePicker.styles';
import TimePicker from '../TimePicker';

describe('Testing TimePicker Component', () => {
  configure({ adapter: new Adapter() });
  let props;

  beforeEach(() => {
    props = {
      label: 'label',
      onChange: sinon.spy(),
    };
  });
  it('renders TimePicker correctly without props', () => {
    const wrapper = shallow(<TimePicker />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('renders TimePicker correctly with props', () => {
    const wrapper = shallow(<TimePicker {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('renders TimePicker correctly with time', () => {
    props.time = '12:00:00';
    const wrapper = shallow(<TimePicker {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should open on Input Focus', () => {
    const wrapper = shallow(<TimePicker {...props} />);
    const dd = wrapper.find(SimpleDropDown).dive();
    const field = dd.find(Field);
    field.simulate('focus');
    expect(wrapper.state('open')).toBeTruthy();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should close on Click Outsidde', () => {
    const wrapper = shallow(<TimePicker {...props} />);
    const co = wrapper.find(ClickOutside);
    wrapper.setState({ open: true });
    co.simulate('outsideClick');
    expect(wrapper.state('open')).toBeFalsy();
  });
  it('should change state and call onChange when update input', () => {
    const wrapper = shallow(<TimePicker {...props} />);
    const dd = wrapper.find(SimpleDropDown).dive();
    const field = dd.find(Field);
    field.simulate('change', '11:30:00');
    expect(wrapper.state('time')).toEqual('11:30:00');
    expect(wrapper.state('hour')).toBe('11');
    expect(wrapper.state('min')).toBe('30');
    expect(wrapper.state('sec')).toBe('00');
    expect(props.onChange.calledWith('11:30:00', '11', '30', '00')).toBeTruthy();
  });
  it('should change state and call onChange when update input with invalid time', () => {
    const wrapper = shallow(<TimePicker {...props} />);
    const dd = wrapper.find(SimpleDropDown).dive();
    const field = dd.find(Field);
    field.simulate('change', '25:61:109');
    expect(wrapper.state('time')).toEqual('23:59:59');
    expect(wrapper.state('hour')).toBe('23');
    expect(wrapper.state('min')).toBe('59');
    expect(wrapper.state('sec')).toBe('59');
    expect(props.onChange.calledWith('23:59:59', '23', '59', '59')).toBeTruthy();
  });
});
