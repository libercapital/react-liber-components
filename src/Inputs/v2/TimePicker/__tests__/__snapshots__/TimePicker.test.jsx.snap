// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing TimePicker Component renders TimePicker correctly with props 1`] = `
<Fragment>
  <GlobalStyleComponent />
  <ClickOutside
    onOutsideClick={[Function]}
  >
    <TimePickerstyles__Container>
      <SimpleDropDown
        autoPosition={true}
        containerSize={false}
        customButton={
          <ForwardRef(TimePickerstyles__Field)
            label="label"
            mask="99:99:99"
            maskChar={null}
            onChange={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            time={null}
            value=""
            version="2"
            width="340px"
          />
        }
        hasOutsideLabel={false}
        label=""
        open={false}
        position="bottom-left-down-right"
        window={[Window]}
      >
        <Clock
          hour=""
          min=""
          onClick={[Function]}
          sec=""
        />
      </SimpleDropDown>
      <TimePickerstyles__Icon
        hide={false}
      />
      <TimePickerstyles__Close
        onClick={[Function]}
        onMouseEnter={[Function]}
        show={false}
      />
    </TimePickerstyles__Container>
  </ClickOutside>
</Fragment>
`;

exports[`Testing TimePicker Component renders TimePicker correctly with time 1`] = `
<Fragment>
  <GlobalStyleComponent />
  <ClickOutside
    onOutsideClick={[Function]}
  >
    <TimePickerstyles__Container>
      <SimpleDropDown
        autoPosition={true}
        containerSize={false}
        customButton={
          <ForwardRef(TimePickerstyles__Field)
            label="label"
            mask="99:99:99"
            maskChar={null}
            onChange={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            time="12:00:00"
            value="12:00:00"
            version="2"
            width="340px"
          />
        }
        hasOutsideLabel={false}
        label=""
        open={false}
        position="bottom-left-down-right"
        window={[Window]}
      >
        <Clock
          hour="12"
          min="00"
          onClick={[Function]}
          sec="00"
        />
      </SimpleDropDown>
      <TimePickerstyles__Icon
        hide={false}
      />
      <TimePickerstyles__Close
        onClick={[Function]}
        onMouseEnter={[Function]}
        show={false}
      />
    </TimePickerstyles__Container>
  </ClickOutside>
</Fragment>
`;

exports[`Testing TimePicker Component renders TimePicker correctly without props 1`] = `
<Fragment>
  <GlobalStyleComponent />
  <ClickOutside
    onOutsideClick={[Function]}
  >
    <TimePickerstyles__Container>
      <SimpleDropDown
        autoPosition={true}
        containerSize={false}
        customButton={
          <ForwardRef(TimePickerstyles__Field)
            label=""
            mask="99:99:99"
            maskChar={null}
            onChange={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            time={null}
            value=""
            version="2"
            width="340px"
          />
        }
        hasOutsideLabel={false}
        label=""
        open={false}
        position="bottom-left-down-right"
        window={[Window]}
      >
        <Clock
          hour=""
          min=""
          onClick={[Function]}
          sec=""
        />
      </SimpleDropDown>
      <TimePickerstyles__Icon
        hide={false}
      />
      <TimePickerstyles__Close
        onClick={[Function]}
        onMouseEnter={[Function]}
        show={false}
      />
    </TimePickerstyles__Container>
  </ClickOutside>
</Fragment>
`;

exports[`Testing TimePicker Component should open on Input Focus 1`] = `
<Fragment>
  <GlobalStyleComponent />
  <ClickOutside
    onOutsideClick={[Function]}
  >
    <TimePickerstyles__Container>
      <SimpleDropDown
        autoPosition={true}
        containerSize={false}
        customButton={
          <ForwardRef(TimePickerstyles__Field)
            label="label"
            mask="99:99:99"
            maskChar={null}
            onChange={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            time={null}
            value=""
            version="2"
            width="340px"
          />
        }
        hasOutsideLabel={false}
        label=""
        open={true}
        position="bottom-left-down-right"
        window={[Window]}
      >
        <Clock
          hour=""
          min=""
          onClick={[Function]}
          sec=""
        />
      </SimpleDropDown>
      <TimePickerstyles__Icon
        hide={false}
      />
      <TimePickerstyles__Close
        onClick={[Function]}
        onMouseEnter={[Function]}
        show={false}
      />
    </TimePickerstyles__Container>
  </ClickOutside>
</Fragment>
`;
